### PR TITLE
CAMS-377 search by division

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

As an attorney, in order to find a case that I want to research, I want search for a case by court division alongside other criteria

# Major Changes

Search now takes the option of a court division selected from a multi-select combo-box.

# Testing/Validation

Test coverage is above 90%

# Notes

Some refactoring was done in a number of areas to improve usability and code.  This affected the cases screen and data verification screens.

### CODE REVIEW

**Comment by amorrow-flexion** [comment]
uncommented e2e database deletion step

File: user-interface/src/case-assignment/AssignAttorneyCasesRow.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        }),
+      )
+      .catch((_reason) => {})
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  return (
+    <TableRow className={inTableTransferMode ? 'in-table-transfer-mode' : ''} {...otherProps}>
```
Comment (btposey) [inline user-interface/src/case-assignment/AssignAttorneyCasesRow.tsx:44]: Remove in table transfer mode?? It isn't working, and we are OK with that. Tech debt.

File: user-interface/src/case-assignment/CaseAssignmentScreen.test.tsx [hunk 2]
```diff
@@ -164,36 +162,32 @@ describe('CaseAssignment Component Tests', () => {
...
-                caseTitle: 'Talia Torres and Tylor Stevenson',
-                dateFiled: '2023-04-04',
-                assignments: ['Joe', 'Sam'],
-              },
-            ],
+    vi.spyOn(Chapter15MockApi, 'get').mockImplementation((_path: string): Promise<ResponseData> => {
+      return Promise.resolve({
+        message: 'not found',
+        count: 0,
+        body: [
```
Comment (btposey) [inline user-interface/src/case-assignment/CaseAssignmentScreen.test.tsx:169]: Replace these with MockData.getCaseBasic() and assignments need to be removed as they are now lazy loaded.

File: user-interface/src/case-assignment/CaseAssignmentScreen.tsx [hunk 3]
```diff
@@ -56,12 +55,14 @@ export const CaseAssignment = () => {
...
     await api
-      .list('/cases')
+      .get<CaseWithAssignments[]>('/cases', { divisionCodes: ['081'] })
       .then((res) => {
-        const chapter15Response = res as Chapter15CaseListResponseData;
-        const assignmentList: Chapter15Type[] = chapter15Response?.body?.caseList ?? [];
-        assignmentList.sort(sortByDate).sort(sortByCaseId);
-        setCaseList(assignmentList);
+        const caseList = res.data;
+        caseList.forEach((bCase) => {
```
Comment (btposey) [inline user-interface/src/case-assignment/CaseAssignmentScreen.tsx:61]: Can we remove this? I know it broke a test before we added this for loop.

File: user-interface/src/lib/components/uswds/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      <ul className="usa-pagination__list">
+        {previous && (
+          <li className="usa-pagination__item usa-pagination__arrow">
+            <PaginationButton
+              id={'previous-results'}
+              onClick={() => {
+                retrievePage({
+                  ...searchPredicate,
+                  offset:
+                    (searchPredicate.offset ?? 0) - (searchPredicate.limit ?? DEFAULT_SEARCH_LIMIT),
```
Comment (btposey) [inline user-interface/src/lib/components/uswds/Pagination.tsx:30]: Replace the default offsets with a constant.

File: backend/functions/azure/functions.ts
Comment (fmaddenflx) [inline backend/functions/azure/functions.ts:1]: for some reason github won't let me approve...

File: user-interface/src/data-verification/DataVerificationScreen.test.tsx [hunk 5]
```diff
@@ -9,9 +8,15 @@ import {
   TransferOrder,
   ConsolidationOrder,
   isConsolidationOrder,
+  Order,
 } from '@common/cams/orders';
 import { OfficeDetails } from '@common/cams/courts';
 import * as FeatureFlagHook from '@/lib/hooks/UseFeatureFlags';
+import { ResponseData } from '@/lib/type-declarations/api';
+
+interface OrderResponseData extends ResponseData<Order> {
```
[the remark this answers is not in the document]
Comment (btposey) [reply user-interface/src/data-verification/DataVerificationScreen.test.tsx:17]: NOT DOING. This API function has not been upgraded to use the new response wrappers.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx [hunk 6]
```diff
@@ -417,10 +410,10 @@ export function ConsolidationOrderAccordion(props: ConsolidationOrderAccordionPr
       };
 
       setIsConsolidationProcessing(true);
-      api
-        .put('/consolidations/approve', data)
+      genericApi
+        .put<ConsolidationOrder[]>('/consolidations/approve', data)
```
[the remark this answers is not in the document]
Comment (btposey) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:414]: No. The consolidations endpoint has not been upgraded to use the common response wrapper.

File: user-interface/src/case-assignment/CaseAssignmentScreen.tsx (hunk 3 above)
Comment (btposey) [reply user-interface/src/case-assignment/CaseAssignmentScreen.tsx:61]: Yeah we broke something before.

File: backend/functions/lib/adapters/gateways/cases.local.gateway.ts [hunk 7]
```diff
@@ -2,11 +2,22 @@ import { CasesInterface } from '../../use-cases/cases.interface';
...
 import { getMonthDayYearStringFromDate } from '../utils/date-helper';
-import { CaseDetail, CaseSummary } from '../../../../../common/src/cams/cases';
+import { CaseBasics, CaseDetail } from '../../../../../common/src/cams/cases';
+import { CamsError } from '../../common-errors/cams-error';
+import { CasesSearchPredicate } from '../../../../../common/src/api/search';
 
 const MODULE_NAME = 'MOCK-CASES-GATEWAY';
 
 export class CasesLocalGateway implements CasesInterface {
+  searchCases(
```
[the remark this answers is not in the document]
Comment (btposey) [reply backend/functions/lib/adapters/gateways/cases.local.gateway.ts:12]: No, we need to consider reviving the new "local" in memory gateway we started.

**Review by btposey — APPROVED** [review]

**Pull request merged**